File: 3_24/3task.cpp
#include <iostream>
using namespace std;

int EvenDigits(int number) {
    if (number < 0) {
        number = -number;
        }
    int count = 0;
    while (number > 0) {
        int digit = number % 10;
        if (digit % 2 == 0) {
            count++;
        }
        number /= 10;
    }
    return count;
}

int main() {
    int n;
    cout << "Enter the number of numbers: ";
    cin >> n;
    if (n>0){
        cout << "Enter the numbers separated by a space: ";
        for (int i = 0; i < n; ++i) {
            int number;
            cin >> number;
            cout << EvenDigits(number);
            if (i < n - 1) {
                cout << " ";
            }
        }
        cout << endl;
    }
    else {
        cout << "Incorrect data" << endl;
    }
    return 0;
}

File: 3_24/3task.rb
def even_digits(number)
    count = 0
    while number > 0
        digit = number % 10
        if digit % 2 == 0
            count += 1
        end
        number /= 10
    end
    return count
end

puts "Enter the number of numbers: "
n = gets.chomp.to_i
puts "Enter the numbers separated by a space: "
numbers = gets.chomp.split.map(&:to_i)

for i in 0...n
    number = numbers[i]
    print even_digits(number)
    if i < n - 1
        print " "
    end
end

puts